contrib/scripts/dumpcerts.sh

```diff
@@ -0,0 +1,141 @@
+#   util-linux
+#   openssl
+#   jq
+# The MIT License (MIT)
+# 
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+# 
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+# 
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# Exit codes:
+# 1 - A component is missing or could not be read
+# 2 - There was a problem reading acme.json
+# 4 - The destination certificate directory does not exist
+# 8 - Missing private key
+
+USAGE="dumpcerts.sh <path to acme> <destination cert directory>"
+
+# Allow us to exit on a missing jq binary
+exitJQ() {
+	echo ""
+	echo "You must have the binary 'jq' to use this."
+	echo "jq is available at: https://stedolan.github.io/jq/download/"
+	echo ""
+	echo ${USAGE}
+	exit 1
+}
+
+
+badACME() {
+	echo ""
+	echo "There was a problem parsing your acme.json file."
+	echo ""
+	echo ${USAGE}
+	exit 2
+}
+
+
+acme="${1}"
+certdir="${2}"
+
+if [ ! -r "${acme}" ]; then
+	echo ""
+	echo "There was a problem reading from '${acme}'"
+	echo "We need to read this file to explode the JSON bundle... exiting."
+	echo ""
+	echo ${USAGE}
+	exit 1
+fi	
+
+
+if [ ! -d "${certdir}" ]; then
+	echo ""
+	echo "Path ${certdir} does not seem to be a directory"
+	echo "We need a directory in which to explode the JSON bundle... exiting."
+	echo ""
+	echo ${USAGE}
+	exit 4
+fi	
+
+jq=$(which jq) || exitJQ
+
+priv=$(${jq} -r '.PrivateKey' ${acme}) || badACME
+
+if [ ! -n "${priv}" ]; then
+	echo ""
+	echo "There didn't seem to be a private key in ${acme}."
+	echo "Please ensure that there is a key in this file and try again."
+	exit 8
+fi
+
+# If they do not exist, create the needed subdirectories for our assets
+# and place each in a variable for later use, normalizing the path
+mkdir -p "${certdir}"/{certs,private}
+
+pdir=$(realpath "${certdir}/private/")
+cdir=$(realpath "${certdir}/certs/")
+
+# Save the existing umask, change the default mode to 600, then
+# after writing the private key switch it back to the default
+oldumask=$(umask)
+umask 177
+# For some reason traefik stores the private key in stripped base64 format, but
```

Comment:
@emilevauge Yes. I think it would be nice to rephrase

_For some reason traefik stores the private key in stripped base64 format_

into something like

_traefik stores the private key in stripped base64 format because [reason]_

or just 

_traefik stores the private key in stripped base64 format_ (without explanation)

I don't think we should leave comments in scripts that express uncertainty about the reasoning in our own code if we can avoid it.